pkg/sql/plan/function/builtin/unary/bin.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+
+import(
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/bin"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"golang.org/x/exp/constraints"
+	"fmt"
+)
+
+func Bin[T constraints.Unsigned | constraints.Signed](vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	inputVector := vectors[0]
+	resultType := types.Type{Oid: types.T_varchar}
+	inputValues := vector.MustTCols[T](inputVector)
+	if inputVector.IsScalar() {
+		if inputVector.ConstVectorIsNull() {
+			return proc.AllocScalarNullVector(resultType), nil
+		}
+		resultValues := make([]string, 1)
+		bin.Bin(inputValues, resultValues)
+		return vector.NewConstString(resultType, inputVector.Length(), resultValues[0]), nil
+	} else {
+		resultValues := make([]string, len(inputValues))
+		bin.Bin(inputValues, resultValues)
```

Comment:
that's not necessary to add a file with some function in vectorize/bin.

I don't know if that's a good idea to use NewWithStrings in some case like lot's of nulls.

Reply:
Best way is to pass in vector and directly writes to vector.   I guess this is more convenient.  Lots of nulls is not a problem, but each string is created then copied into vector.   This can be a noticeable performance penalty.

Reply:
I will try to pass in vector in the function